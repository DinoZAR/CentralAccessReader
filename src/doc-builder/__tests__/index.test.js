import DocBuilder from '../index';
import fs from 'fs';

jest.mock('fs');

describe('DocBuilder()', () => {
  it('builds a paragraph and writes to disk', () => {
    const builder = new DocBuilder('tmp');
    builder.build((build) => {
      build.paragraph();
      build.text({ content: 'Some text' });
      build.math({ mathml: '<mathml>x</mathml>' });
      build.image({ url: 'image.png' });
    });
    expect(fs.writeFile.mock.calls).toHaveLength(1);
    expect(fs.writeFile.mock.calls[0][0]).toBe('tmp/__chunks/0.json');
    expect(fs.writeFile.mock.calls[0][1]).toMatchSnapshot();
  });
});
