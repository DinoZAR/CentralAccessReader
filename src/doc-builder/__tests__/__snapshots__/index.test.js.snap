// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocBuilder() builds a paragraph and writes to disk 1`] = `"[{\\"id\\":1,\\"type\\":\\"paragraph\\",\\"content\\":[{\\"id\\":2,\\"type\\":\\"text\\",\\"content\\":\\"Some text\\"},{\\"id\\":3,\\"type\\":\\"math\\",\\"mathml\\":\\"<mathml>x</mathml>\\"},{\\"id\\":4,\\"type\\":\\"image\\",\\"url\\":\\"image.png\\"}]}]"`;
